### DISCUSSION TITLE
stream_select_ptr info

### DISCUSSION THREAD

**Discussion by mfp20** [grblHAL/core#31] [Q&A] [answered]
"Optional pointer to function for switching between I/O streams."

Is it used to select where to read/write console? I'm asking because it could be useful for me to be able to use i2c/spi/usb instead of uart.

**Comment by terjeio** [comment, marked as answer]
Yes, but it is not needed for switching if the switch is permanent. Currently this function is only implemented in a few drivers - those that support a network connection. If not available replace the other entry points in your initialization code.

FYI I have a few changes coming in the next build that allows more flexible use/handling of streams.

**Reply by mfp20** [reply]
Yeah, it should be permanent. In any case the console is a io_stream_t, so I can use any I/O I want, as long as I turn it into a io_stream_t . Right?

**Reply by terjeio** [reply]
Correct, for core use only non-optional functions needs to be implemented.
